src/main/java/edu/wpi/grip/ui/preview/PreviewsView.java

```diff
@@ -0,0 +1,69 @@
+package edu.wpi.grip.ui.preview;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.Socket;
+import edu.wpi.grip.core.events.SocketPreviewChangedEvent;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A simple JavaFX container that automatically shows previews of all sockets marked as "previewed".
+ *
+ * @see Socket#isPreviewed()
+ */
+public class PreviewsView extends VBox {
+
+    @FXML
+    private HBox previewBox;
+
+    private final EventBus eventBus;
+    private final List<OutputSocket<?>> previewedSockets;
```

Comment:
Might be better to make this a map of `OutputSocket` to `SocketPreviewView`
That would allow you to remove the exact objects instead of assuming that the indexes will line up between the two lists.
See below...